layers/+lang/lua/funcs.el

```diff
@@ -33,19 +42,34 @@
 
 
 ;; LSP Lua
-(defun spacemacs//lua-setup-lsp-emmy ()
-  "Setup LSP Lua."
+(defun spacemacs//lua-setup-lsp-emmy-lua ()
+  "Setup emmy Lua."
   (when lua-lsp-emmy-java-path
     (setq lsp-clients-emmy-lua-java-path lua-lsp-emmy-java-path))
   (when lua-lsp-emmy-jar-path
     (setq lsp-clients-emmy-lua-jar-path (expand-file-name lua-lsp-emmy-jar-path)))
   (setq lsp-enable-file-watchers lua-lsp-emmy-enable-file-watchers)
+  (set (make-local-variable 'lsp-disabled-clients) '(lua-language-server lsp-lua-lsp))
+  (lsp))
+
+(defun spacemacs//lua-setup-lsp-lua-language-server ()
+  "Setup lua-language-server."
+  (set (make-local-variable 'lsp-disabled-clients) '(emmy-lua lsp-lua-lsp))
   (lsp))
 
+(defun spacemacs//lua-setup-lsp-lua-lsp ()
+  "Setup lsp-lua-lsp."
+  (set (make-local-variable 'lsp-disabled-clients) '(emmy-lua lsp-lua-language-server))
+  (lsp))
+
+(defun spacemacs//lua-setup-lsp-lua ()
+  "Setup lsp for lua"
+  (lsp))
 
 ;; Lua mode
 (defun spacemacs//lua-setup-company-lua ()
   (spacemacs|add-company-backends
     :backends company-lua
-    :modes lua-mode)
```

Comment:
It's not clear to me why these changes are made.

Reply:
resolved in new submition.